### PR TITLE
feat: scrollable nav

### PR DESCRIPTION
**Pull request by tim08094495757**
## Types of changes
* **New feature**

## Description
blocked by #218
1. make navbar scrollable

## Steps to Test This Pull Request
1. set screen width between 768px and 1193px
2. click arrow

## Expected behavior
image

### CODE REVIEW

**Review by flynnhou — CHANGES_REQUESTED** [review]
@tim08094495757 Please rebase `scrollable-nav-bar` with your `main` branch while eliminating ffca8fb6542c4ded2c9586db64a23d270a1c47a9.

## Why

Commit ffca8fb6542c4ded2c9586db64a23d270a1c47a9 is a condensed change of the entire #218 content. But it doesn't explain as clearly as it should in every commit message in that PR.
Once #218 is merged, if you rebase your `scrollable-nav-bar` with your `main`, this PR will only show the necessary two commits after you update your `scrollable-nav-bar` with the merged origin `main` branch.
This way, it's more clear on what to review, it provides contributors with the correct scope of changes, and it tracks the dependent commits in #218's branch with the latest updates.

## How

As there's never only one correct answer, the following steps are just for you to consider as a reference.

### Before #218 is merged

- Pull `tim08094495757/pycontw-frontend`'s `main` for your local `main`
  ```bash
  git pull <remote_name> main
  ```
- Check out to `scrollable-nav-bar`
  ```bash
  git checkout scrollable-nav-bar
  ```
- Rebase with your latest `main`
  ```bash
  git rebase main
  ```
- Push to remote
  ```bash
   git push -f
  ```

### After #218 is merged

- Pull `pycontw/pycontw-frontend`'s `main`
  ```bash
  git pull <remote_name> main
  ```
- Rebase
  ```bash
  git checkout scrollable-nav-bar
  git rebase <remote_name>/main
  ```
- Push
  ```bash
  git push -f
  ```

---

If there's any question, please contact me for clarifications.

File: components/core/header/nav-bar/ScrollableNavBar.vue
Comment (flynnhou) [inline components/core/header/nav-bar/ScrollableNavBar.vue:75]: 1. Use proper camelCase to avoid variable name typos - `newScrollLeft`
2. If `el?.scrollLeft` evaluates as `undefined`, then is the value of `newScrollLeft` `undefined` or `NaN`?

File: components/core/header/nav-bar/ScrollableNavBar.vue
Comment (flynnhou) [inline components/core/header/nav-bar/ScrollableNavBar.vue:108]: Could Tailwind's widths replace this?

File: components/core/header/nav-bar/ScrollableNavBar.vue
Comment (flynnhou) [inline components/core/header/nav-bar/ScrollableNavBar.vue:117]: Remove unnecessary `px` after `0`

**Comment by mattwang44** [comment]
- #218 has been merged
- I just rebased this branch to the latest main branch

File: components/core/header/nav-bar/ScrollableNavBar.vue
Comment (tim08094495757) [reply components/core/header/nav-bar/ScrollableNavBar.vue]: #88493a9

File: components/core/header/nav-bar/ScrollableNavBar.vue
Comment (tim08094495757) [reply components/core/header/nav-bar/ScrollableNavBar.vue]: 88493a9

File: components/core/header/nav-bar/ScrollableNavBar.vue
Comment (tim08094495757) [reply components/core/header/nav-bar/ScrollableNavBar.vue]: 88493a9

File: components/core/header/Header.vue
Comment (flynnhou) [inline components/core/header/Header.vue:70]: Minor suggestion that doesn't block the PR:

```suggestion
     /* enable click-through ScrollableNavBar by `pointer-events: auto` */
    @apply w-full flex justify-between items-center px-4 pointer-events-auto;
```

It looks verbose but using tailwind for consistency is better in my opinion

File: components/core/header/nav-bar/ScrollableNavBar.vue
Comment (flynnhou) [inline components/core/header/nav-bar/ScrollableNavBar.vue:72]: Because we don't have strict ESLint settings, this is just a comment to point out a better practice.

Normally, in JavaScript, we don't reassign the parameter to avoid unexpected side effects. Instead, we make a new variable, e.g.,

```js
const predicate = scrollLeft || el.scrollLeft
return predicate <= 0
```

> Just a quick example, not a suggestion to change

After all, the methods are just doing a read-only kind of operation. In an edge case, if there's a global variable passed in, it'll become a [side effect](https://en.wikipedia.org/wiki/Side_effect_%28computer_science%29) which is unnecessary here.

File: components/core/header/nav-bar/ScrollableNavBar.vue
Comment (flynnhou) [inline components/core/header/nav-bar/ScrollableNavBar.vue:88]: Similar here. Not going to block but just a friendly comment.

1. In JavaScript, early-return `if` statement style is more widely adopted. I.e.:
   ```js
   if (!el || !newScrollLeft) {
     return
   }
   // Logic here
   ```
   It avoids spaghetti if-else statements and is easier to read. Nevertheless, it's still a choice rather than a rule.
1. The `const newScrollLeft = el && el.scrollLeft + delta` still smells bad but I can't give a concrete suggestion at this point.

**Review by flynnhou — APPROVED** [review]
@tim08094495757 I've tested this on Safari and it looks as good as the other browsers.
I wanted to implement this feature last year but it didn't happen. So much appreciated!

LGTM!

File: components/core/header/nav-bar/ScrollableNavBar.vue
Comment (tim08094495757) [reply components/core/header/nav-bar/ScrollableNavBar.vue:72]: Because I try avoid the case like 0 || el.scrollLeft. But thanks for the good suggestion. Helped me remind the expression like 0 ?? el.scrollLeft. It's much cleaner, haha.

File: components/core/header/Header.vue
Comment (tim08094495757) [reply components/core/header/Header.vue:70]: Thanks. I will use this style in the future.

File: components/core/header/nav-bar/ScrollableNavBar.vue
Comment (tim08094495757) [reply components/core/header/nav-bar/ScrollableNavBar.vue:88]: 2. agreed~

**Comment by tim08094495757** [comment]
> @tim08094495757 I've tested this on Safari and it looks as good as the other browsers. I wanted to implement this feature last year but it didn't happen. So much appreciated!
> 
> LGTM!

It's very helpful. Thank you so much.

**Pull request merged**
